src/variants/lambdaregistry/implementation.mligo

```diff
@@ -0,0 +1,160 @@
+// operations.
+// -----------------------------------------------------------------
+
+let decision_callback (input : decision_callback_input)
+    : decision_callback_output = let
+  (ops, new_ce, new_guardian) = match unpack_proposal_metadata(input.proposal.metadata) with
+    | Add_handler ahp -> (([] : operation list), add_handler(input.extras, ahp), None)
+    | Remove_handler rhp -> (([] : operation list), remove_handler(input.extras, rhp), None)
+    | Execute_handler ehp -> execute_handler(input.proposal.proposer, input.proposal.proposer_frozen_token, input.extras, input.proposal.metadata, ehp)
+  in { operations = ops; extras = new_ce; guardian = new_guardian }
+
+// -----------------------------------------------------------------
+// Rejected proposal slash value is called when a proposal is rejected, and the
+// value that voters get back can be slashed. This procedure should return the
+// amount to be slashed.
+// -----------------------------------------------------------------
+
+let rejected_proposal_slash_value (params, ce : proposal * contract_extra) : nat =
+  let slash_scale_value = fetch_nat("slash_scale_value", ce.handler_storage) in
+  let slash_division_value = fetch_nat("slash_division_value", ce.handler_storage)
+  in (slash_scale_value * params.proposer_frozen_token) / slash_division_value
+
+// -----------------------------------------------------------------
+// Type that make up the custom entrypoint of the variant.
+//
+// -----------------------------------------------------------------
+
+type lookup_registry_param =
+  [@layout:comb]
+  { key : registry_key
+  ; callback : address
+  }
+
+type custom_ep_param =
+  | Lookup_registry of lookup_registry_param
+  | RegistryCepDummy of unit
+
+type lookup_registry_view = (registry_key * (registry_value option)) contract
+
+// A custom entrypoint to fetch values from Registry
+let lookup_registry (param, store : lookup_registry_param * storage) : operation list * storage =
+  let view_contract : lookup_registry_view =
+      match (Tezos.get_contract_opt(param.callback) : lookup_registry_view option) with
+      | Some callback_contract -> callback_contract
+      | None -> (failwith bad_view_contract: lookup_registry_view) in
+  let contract_extra = store.extra in
+  let registry : registry = fetch_registry(contract_extra.handler_storage) in
+  let value_at_key : registry_value option = Map.find_opt param.key registry in
+  let operation : operation = Tezos.transaction (param.key, value_at_key) 0mutez view_contract
+  in ((operation :: nil_op), store)
+
+// -----------------------------------------------------------------
+// Custom entrypoint handler for this variant
+//
+// -----------------------------------------------------------------
+let custom_ep (custom_ep_param, storage : custom_ep_param * storage): operation list * storage
+  = match custom_ep_param with
+  | Lookup_registry p -> lookup_registry(p, storage)
+  | _ -> (([]: operation list), storage)
```

Comment:
From what I understand, we need to have different variant to support the custom entrypoints. This is probably not intended since if the user want custom entrypoints, he/she should use compile version of BaseDAO instead.

I think either we don't support custom entrypoints in LambdaDAO for now or find another way to support custom entrypoints in a dynamic way (user may be able to create/remove) later on.

Reply:
Yes, LambdaDAO itself does not have a way to accomodate custom entrypoints. But LambdaDAO, being a variant itself, can be modified to have any custom entrypoint that is required.

Reply:
> But LambdaDAO, being a variant itself

I suggest to not have a variant of LambdaDAO and simply not support custom entrypoints for now. 

The main reason is that, when using LambdaDAO, to have treasury or registry dao, one simply just update the handler without having to deploy another contract. 

If we have variant of LambdaDAO for treasury or registry, it seems to defeat the purpose, because then one have to deploy a separate contract.

@pasqu4le may have a different idea.

Reply:
Do you mean that we should not include the "lookup_registry" custom entrypoint for registry variant of LambdaDAO? If yes, do you have a suggestion on how to include it?

Reply:
Yep, I think we simply state we don't support custom entrypoint for now. 

If this is needed as it should be, we should figure out another way to create custom entrypoints later on, perhaps allowing the ability to create/remove custom entrypoints similarly to how the handler work. 

I guess, we may have to wait for Pasquale feedback.

Reply:
>Yep, I think we simply state we don't support custom entrypoint for now.

But we don't, and the 'custom' entrypoint for registry variant is not really dynamic. It is included because it is required to complete the registryDAO functionality.

Reply:
> I suggest to not have a variant of LambdaDAO and simply not support custom entrypoints for now.

I think that's not possible, one requirement for this variant is the ability to define the other existing ones (`registry` and `treasury`) within it, and since they have some custom entrypoints they are needed here too.

> If we have variant of LambdaDAO for treasury or registry, it seems to defeat the purpose, because then one have to deploy a separate contract.

I think that the point is to be able to tweak the behavior of a DAO by changing the available handlers, but not change it drastically, so choosing the (fixed) custom entrypoint once should be fine.

> one simply just update the handler without having to deploy another contract.

those are two different things however, custom entrypoints can be called directly, whereas the available handlers can only be used for proposals (following the same cycle of voting and `flush`ing as every other proposal).